tasks/upload-sbom-to-atlas/upload-sbom-to-atlas.yaml

```diff
@@ -0,0 +1,261 @@
+      script: |
+        #!/usr/bin/env bash
+        set -o errexit -o pipefail -o nounset
+        
+        skipUploadingSbom=$(params.skipUploadingSbom)
+        if [[ "$skipUploadingSbom" == "true" ]]; then
+          echo "Skipping SBOM uploading - skipUploadingSbom == true"
+          exit 0
+        fi
+
+        version_lesser_equal() {
+          local first
+          first="$(printf "%s\n%s" "$1" "$2" | sort --version-sort | head -n 1)"
+        
+          if [ "$1" = "$first" ]; then
+            echo "SBOM version ($1) is supported (<= $2), will not convert"
+          else
+            echo "SBOM version ($1) is not supported, will convert to $2"
+            printf "%s" "$2" > "${sbom_path}.convert_to_version"
+          fi
+        }
+        
+        echo "Looking for CycloneDX or SPDX SBOMs in $sbomsDir"
+        
+        find "$sbomsDir" -type f | while read -r filepath; do
+          # Skip non-JSON files
+          file_relpath=$(realpath "$filepath" --relative-base="$sbomsDir")
+          if ! jq empty "$filepath" 2>/dev/null; then
+            echo "$file_relpath: not a valid JSON"
+            continue
+          fi
+        
+          # Extract the format information using jq
+          sbom_format=$(jq -r 'if .bomFormat then .bomFormat else if .spdxVersion then "SPDX" else empty end end' "$filepath")
+        
+          # If sbom_format is empty, it's not a valid SBOM (CycloneDX or SPDX)
+          if [ -z "$sbom_format" ]; then
+            echo "$file_relpath: not a valid SBOM (CycloneDX or SPDX)"
+            continue  # Skip further processing for non-SBOM files
+          fi
+        
+          # The 'id' of each SBOM is checksum of the original content, before (possibly)
+          # downgrading the CycloneDX version. The conversion always updates some metadata
+          # (timestamp, UUID), changing the checksum. To avoid duplication, use the original
+          # checksum.
+          echo "Calculating SBOM ID for $file_relpath"
+          sbom_id="sha256:$(sha256sum "$filepath" | cut -d ' ' -f 1)"
+        
+          echo "Found $sbom_format SBOM: $file_relpath"
+        
+          # Symlink the discovered SBOMS to ${workdir}/${sbom_id}.json so that subsequent steps
+          # don't have to look for them again.
+          sbom_path="$workdir/$sbom_id.json"
+          ln -s "$(realpath "$filepath")" "$sbom_path"
+        
+          # Check if SBOM is CycloneDX or SPDX
+          if [ "$sbom_format" == "CycloneDX" ]; then
+            echo "Found CycloneDX SBOM: $file_relpath"
+        
+            sbom_version="$(jq -r ".specVersion" "$sbom_path")"
+            supportedCycloneDxVersion=$(params.supportedCycloneDxVersion)
+            version_lesser_equal "$sbom_version" "$supportedCycloneDxVersion"
+        
+            printf "%s" "cyclonedx-json" > "${sbom_path}.sbom_format"
+        
+          elif [ "$sbom_format" == "SPDX" ]; then
+            echo "Found SPDX SBOM: $file_relpath"
```

Comment:
The same is already printed on line 123.

Reply:
Good catch, thanks!